test/cpp_extensions/aes.cuh

```diff
@@ -0,0 +1,153 @@
+  0x33, 0x66, 0xcc, 0x83, 0x1d, 0x3a, 0x74, 0xe8, 0xcb, 0x8d, 0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 
+  0x40, 0x80, 0x1b, 0x36, 0x6c, 0xd8, 0xab, 0x4d, 0x9a, 0x2f, 0x5e, 0xbc, 0x63, 0xc6, 0x97, 0x35, 
+  0x6a, 0xd4, 0xb3, 0x7d, 0xfa, 0xef, 0xc5, 0x91, 0x39, 0x72, 0xe4, 0xd3, 0xbd, 0x61, 0xc2, 0x9f, 
+  0x25, 0x4a, 0x94, 0x33, 0x66, 0xcc, 0x83, 0x1d, 0x3a, 0x74, 0xe8, 0xcb, 0x8d, 0x01, 0x02, 0x04, 
+  0x08, 0x10, 0x20, 0x40, 0x80, 0x1b, 0x36, 0x6c, 0xd8, 0xab, 0x4d, 0x9a, 0x2f, 0x5e, 0xbc, 0x63, 
+  0xc6, 0x97, 0x35, 0x6a, 0xd4, 0xb3, 0x7d, 0xfa, 0xef, 0xc5, 0x91, 0x39, 0x72, 0xe4, 0xd3, 0xbd, 
+  0x61, 0xc2, 0x9f, 0x25, 0x4a, 0x94, 0x33, 0x66, 0xcc, 0x83, 0x1d, 0x3a, 0x74, 0xe8, 0xcb
+};
+
+__device__ void addRoundKey(uint8_t* block, uint8_t* key){
+  for (auto i = 0; i < 16; i++) {
+    block[i] = block[i] ^ key[i];
+  }
+}
+
+__device__ void mixColumns(uint8_t* block){
+  for (auto i = 0; i < 4; i++) {
+    uint8_t x[4];
+    uint8_t y[4];
+    for (auto j = 0; j < 4; j++) {
+      x[j] = block[4 * i + j];
+      uint8_t z = ((int8_t)x[j] >> 7);
+      y[j] = x[j] << 1;
+      y[j] ^= 0x1b & z;
+    }
+    block[4 * i + 0] = y[0] ^ x[3] ^ x[2] ^ y[1] ^ x[1];
+    block[4 * i + 1] = y[1] ^ x[0] ^ x[3] ^ y[2] ^ x[2];
+    block[4 * i + 2] = y[2] ^ x[1] ^ x[0] ^ y[3] ^ x[3];
+    block[4 * i + 3] = y[3] ^ x[2] ^ x[1] ^ y[0] ^ x[0]; 
+  }
+}
+
+__device__ void subBytes(uint8_t* block){
+  for (auto i = 0; i < 16; i++) {
+    block[i] = sbox[block[i]];
+  }
+}
+
+__device__ void shiftRows(uint8_t* block) {
+  uint8_t temp;
+
+  temp = block[1];
+  block[1] = block[5];
+  block[5] = block[9];
+  block[9] = block[13];
+  block[13] = temp;
+
+  temp = block[2];
+  block[2] = block[10];
+  block[10] = temp;
+
+  temp = block[6];
+  block[6] = block[14];
+  block[14] = temp;
+
+  temp = block[3];
+  block[3] = block[15];
+  block[15] = block[11];
+  block[11] = block[7];
+  block[7] = temp;
+}
+
+__device__ void expandKey(const uint8_t* key, uint8_t* rkey){
+  uint8_t temp[4];
+  uint32_t n = 10;
+  for (auto i = 0; i < 4; ++i) {
+    rkey[4 * i] = key[4 * i];
+    rkey[4 * i + 1] = key[4 * i + 1];
+    rkey[4 * i + 2] = key[4 * i + 2];
+    rkey[4 * i + 3] = key[4 * i + 3];
+  }
+  for (auto i = 4; i < 4 * (n + 1); i++) {
+    for (auto j = 0; j < 4; j++) {
+      temp[j] = rkey[j + 4 * (i - 1)];
+    }
+    if (i % 4 == 0) {
+      auto t = temp[0];
+      temp[0] = temp[1];
+      temp[1] = temp[2];
+      temp[2] = temp[3];
+      temp[3] = t;
+
+      temp[0] = sbox[temp[0]];
+      temp[1] = sbox[temp[1]];
+      temp[2] = sbox[temp[2]];
+      temp[3] = sbox[temp[3]];
+  
+      temp[0] = temp[0] ^ rcon[i / 4];
+    }
+    rkey[4 * i] = temp[0] ^ rkey[4 * (i - 4)];
+    rkey[4 * i + 1] = temp[1] ^ rkey[4 * (i - 4) + 1];
+    rkey[4 * i + 2] = temp[2] ^ rkey[4 * (i - 4) + 2];
+    rkey[4 * i + 3] = temp[3] ^ rkey[4 * (i - 4) + 3];
+  } 
+}
+
+__device__ void encrypt(uint8_t* block, const uint8_t* key) {
+  uint8_t rkey[176];
```

Comment:
you probably want to make consts out of 10 and 16 and derive 176 = 16 * (10 + 1)